hw5_iterator/main.py

```diff
@@ -0,0 +1,243 @@
+'''
+Homework #5 - Iterators and Generators.
+'''
+
+import random
+
+
+# -----------------------------------------------------------------------------
+# Task 1 - Generator to read fasta file.
+# -----------------------------------------------------------------------------
+
+def fasta_reader(file_path: str):
+    '''
+    This function is a generator to read FASTA file.
+    Parameter:
+        file_path - path to FASTA file.
+    Yield values:
+        idx - sequence ID
+        seq - sequence body
+    '''
+    f = open(file_path)
+    prev_id = None              # storage id-line between sequences reading
+    while True:
+        # looking for ID string '>...'
+        if not prev_id:
+            idx = f.readline()
+        else:
+            idx = prev_id
+            prev_id = None
+        if not idx:
+            break
+        if idx[0] != '>':
+            continue
+        idx = idx.strip()
+        # get next strings with sequence, seq can be several lines
+        seq = ''
+        while True:
+            s = f.readline()
+            if not s:
+                break
+            s = s.strip()
+            if s[0] != '>':
+                seq += s        # add line to sequence
+            else:
+                prev_id = s     # store id-line for the next sequence reading
+                break
+        if not seq:
+            break
+        yield idx, seq
+
+
+def task1_test():
+
+    reader = fasta_reader('sequences.fasta')
+    print(type(reader))
+
+    for id_, seq in reader:
+        print(id_, seq[:50])
+
+
+# -----------------------------------------------------------------------------
+# Task 2 - Class for reading sequences with small changes.
+# -----------------------------------------------------------------------------
+
+class FastaChangingReader:
+    '''
+    This class is iterator that every call reads the next sequence
+    from file, makes small random changes of sequence and returns
+    two values: id and seq.
+    '''
+    def __init__(self, file_path: str):
+        '''
+        Parameter:
+            file_path - path to FASTE file.
+        '''
+        self.file_path = file_path
+        self.gen = fasta_reader(file_path)
```

Comment:
Этот атрибут я бы сделал приватным
```suggestion
        self._gen = fasta_reader(file_path)
```